force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -124,18 +147,20 @@ public with sharing class ClassesObjectsMethods {
     public static String findTheScientist() {
         // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
+        String scientistFound = '';
+        String letterToMatch = 'g';
+        
 
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        for(String scientist : scientistNames){
+            if(scientist.toLowerCase().contains(letterToMatch)){
+                scientistFound += scientist + ',';
+            }else {
+                continue;
+            }
+        }
+        String scientistWithG = scientistFound.removeEnd(',');
+        return scientistWithG;
```

Comment:
Now consider doing this problem by just creating a List of `scientistNames`, collecting each scientistName, and using String.join to combine them into one csv.